includes/ucf-charts-posttype.php

```diff
@@ -0,0 +1,273 @@
+<?php
+/**
+ * Handles registration of the ucf-chart custom post type
+ **/
+if ( ! class_exists( 'UCF_Chart_PostType' ) ) {
+	class UCF_Chart_PostType {
+		/**
+		 * Registers the ucf-chart custom post type
+		 * @author Jim Barnes
+		 * @since 1.0.0
+		 **/
+		public static function register() {
+			$labels = apply_filters(
+				'ucf_chart_labels',
+				array(
+					'singular'    => 'Chart',
+					'plural'      => 'Charts',
+					'text_domain' => 'ucf_chart'
+				)
+			);
+
+			register_post_type( 'ucf_chart', self::args( $labels ) );
+
+			if ( UCF_Chart_Config::get_option_or_default( 'include_fields' ) ) {
+				add_action( 'add_meta_boxes', array( 'UCF_Chart_PostType', 'register_metaboxes' ) );
+				add_action( 'save_post', array( 'UCF_Chart_PostType', 'save_metaboxes' ) );
+			}
+		}
+
+		/**
+		 * Registers the ucf_chart metaboxes
+		 * @author Jim Barnes
+		 * @since 1.0.0
+		 **/
+		public static function register_metaboxes() {
+			add_meta_box(
+				'ucf_chart_metabox',
+				'UCF Chart Fields',
+				array( 'UCF_Chart_PostType', 'register_fields' ),
+				'ucf_chart',
+				'normal',
+				'low'
+			);
+		}
+
+		/**
+		 * Prints out the fields for the ucf_chart metabox
+		 * @author Jim Barnes
+		 * @since 1.0.0
+		 * @param WP_POST $post The WordPress post object for the current post being edited.
+		 * @return string The html output for the ucf_chart_metabox
+		 **/
+		public static function register_fields( $post ) {
+			wp_nonce_field( 'ucf_chart_nonce_save', 'ucf_chart_nonce' );
+			$upload_link = esc_url( get_upload_iframe_src( 'media', $post->ID ) );
+
+			$chart_types = UCF_Chart_Common::get_chart_types();
+
+			$chart_type       = get_post_meta( $post->ID, 'ucf_chart_type', TRUE );
+			$data_json        = get_post_meta( $post->ID, 'ucf_chart_data_json', TRUE );
+			$data_json_url    = wp_get_attachment_url( $data_json );
+			$options_json     = get_post_meta( $post->ID, 'ucf_chart_options_json', TRUE );
+			$options_json_url = wp_get_attachment_url( $options_json );
+
+			// Existing asset IDs are invalid if the attachment URL can't be retrieved
+			// (i.e. if the attachment was deleted).
+			if ( ! $data_json_url ) {
+				$data_json = null;
+			}
+
+			if ( ! $options_json_url ) {
+				$options_json = null;
+			}
+?>
+			<table class="form-table">
+				<tbody>
+					<tr>
+						<th><strong>Chart Type</strong></th>
+						<td>
+							<select class="meta-select-field" id="ucf_chart_type" name="ucf_chart_type">
+							<?php foreach( $chart_types as $key => $val ) : ?>
+								<option value="<?php echo $key; ?>"<?php echo ( $chart_type === $key ) ? ' selected' : ''; ?>><?php echo $val; ?></option>
+							<?php endforeach; ?>
+							</select>
+						</td>
+					</tr>
+					<tr>
+						<th><strong>Chart Data JSON</strong></th>
```

Comment:
Since this plugin requires the user to upload json files, a upload_mimes hook should be added to ensure they're able to be uploaded.  (Main site already has this hook which is why it probably works out-of-the-box for you locally).  See https://github.com/UCF/Main-Site-Theme/blob/rc-v3.0.0/includes/config.php#L270-L280